packages/asset-swapper/src/utils/market_operation_utils/index.ts

```diff
@@ -150,6 +150,23 @@ export class MarketOperationUtils {
             quoteSourceFilters.isAllowed(ERC20BridgeSource.Balancer),
         );
 
+        const {
+            onChain: sampleCreamOnChain,
+            offChain: sampleCreamOffChain,
+        } = this._sampler.creamPoolsCache.howToSampleCream(
+            takerToken,
+            makerToken,
+            quoteSourceFilters.isAllowed(ERC20BridgeSource.Cream),
+        );
+
+        let excludedSources: ERC20BridgeSource[] = [];
+        if (!sampleCreamOnChain) {
+            excludedSources = excludedSources.concat(ERC20BridgeSource.Cream);
+        }
+        if (!sampleBalancerOnChain) {
+            excludedSources = excludedSources.concat(ERC20BridgeSource.Balancer);
+        }
```

Comment:
idk, what do you think of?
```suggestion
        excludedSources = [
            ...(sampleCreamOnChain ? [ERC20BridgeSource.Cream] : []),
            ...(sampleBalancerOnChain ? [ERC20BalancerSource.Balancer] : []),
        ];
```

Reply:
If not, we should change the `concat()` to `push()`.

Reply:
Great suggestions!